tests/validation/tests/v3_api/test_cluster_templates.py

```diff
@@ -0,0 +1,448 @@
+    members = [{
+        "type": "member",
+        "accessType": "owner",
+        "userPrincipalId": principalid
+    }]
+
+    cluster_template = client.update(cluster_template,
+                                     name=template_name,
+                                     members=members)
+    # As an owner of the template create a revision using the template
+    # and also create a cluster using the template revision
+    revision_name = random_test_name("revision1")
+    cluster_template_revision = \
+        standard_newuser_client.create_cluster_template_revision(
+            name=revision_name,
+            clusterConfig=cluster_config1,
+            clusterTemplateId=cluster_template.id)
+    cluster_name = "test-auto"
+    cluster = create_node_cluster(
+        standard_newuser_client, name=cluster_name,
+        clusterTemplateRevisionId=cluster_template_revision.id)
+
+    cluster_cleanup(standard_newuser_client, cluster)
+
+
+def test_cluster_template_add_readonly_member():
+
+    cluster_config1 = get_cluster_config(RANCHER_DEFAULT_K8S_VERSION)
+    client = get_admin_client()
+
+    standard_newuser_client = Client(
+                                url=CATTLE_API_URL,
+                                token=user_token["newuser_standard"]["token"],
+                                verify=False)
+    # As an Admin, create a cluster template, revision and update the members
+    # list with the new user as read-only user
+    template_name = random_test_name("usertemplate")
+    cluster_template = client.create_cluster_template(
+        name=template_name, description="test-template")
+
+    revision_name = random_test_name("revision1")
+    cluster_template_revision1 = client.create_cluster_template_revision(
+        name=revision_name,
+        clusterConfig=cluster_config1,
+        clusterTemplateId=cluster_template.id)
+
+    principalid = user_token["newuser_standard"]["user"]["principalIds"]
+    members = [{
+        "type": "member",
+        "accessType": "read-only",
+        "userPrincipalId": principalid
+    }]
+
+    cluster_template = client.update(cluster_template,
+                                     name=template_name, members=members)
+
+    # As a read-only member of the rke template verify that
+    # adding another revision to the template fails
+    revision_name = "userrevision"
+    with pytest.raises(ApiError) as e:
+        standard_newuser_client.create_cluster_template_revision(
+            name=revision_name,
+            clusterConfig=cluster_config1,
+            clusterTemplateId=cluster_template.id)
+
+    assert e.value.error.status == 403
+    assert e.value.error.code == 'PermissionDenied'
+
+    # Verify that the read-only user can create a cluster with the existing
+    # template revision
+    cluster_name = "test-auto"
+    cluster = create_node_cluster(
+        standard_newuser_client, name=cluster_name,
+        clusterTemplateRevisionId=cluster_template_revision1.id)
+
```

Comment:
As admin Create another CT with 1 CTR and make sure that "standard_newuser_client" is not able deploy a cluster with existing CTR.

Reply:
Done